google/generativeai/types/permission_types.py

```diff
@@ -233,3 +245,165 @@ async def get_async(
         get_perm_response = await client.get_permission(request=get_perm_request)
         get_perm_response = type(get_perm_response).to_dict(get_perm_response)
         return cls(**get_perm_response)
+
+
+class PermissionAdapter:
```

Comment:
> We'd need to leave deprecated stubs on "Corpus".

Would it make sense to have a `deprecated` decorator on these methods to raise DeprecationWarning with a clear message and the method itself will call the new `.permissions` property for backward compatibility?

Reply:
That would make it strictly backwards compatible. Is it worth it?

Reply:
I was thinking of removing those methods entirely in future releases. First, raise a warning saying that it will be deprecated in the future use this instead and eventually remove it(subsequent releases). But I'm not sure if this is a good practice.

What is the best way to handle it?

Reply:
> But I'm not sure if this is a good practice.

That is good practice, I'm just being sloppy. Go for it if you want, we can merge this whenever you're ready.

Reply:
Done.